Failed to build `stratum-bf`

**Describe the bug**
`stratum-bf` build fails with error in `bf_switch.cc`

**Environment**
Jenkins

Switch:
N/A

**To Reproduce**
Steps to reproduce the behavior:
1. Run build-stratum-bf-container.sh script from stratum/hal/bin/barefoot/docker
2. Build fails with this error:
```stratum/hal/lib/barefoot/bf_switch.cc: In member function 'virtual util::Status stratum::hal::barefoot::BFSwitch::RetrieveValue(stratum::uint64, const stratum::hal::DataRequest&, stratum::hal::WriterInterface<stratum::hal::DataResponse>*, std::vector<util::Status>*)':

stratum/hal/lib/barefoot/bf_switch.cc:233:32: error: 'class util::StatusOr<stratum::hal::DataResponse>' has no member named 'mutable_node_info'

         auto* node_info = resp.mutable_node_info();

                                ^~~~~~~~~~~~~~~~~
```

--- a/stratum/hal/lib/barefoot/bf_chassis_manager.h
+++ b/stratum/hal/lib/barefoot/bf_chassis_manager.h
@@ -70,6 +70,9 @@ class BFChassisManager {
   ::util::StatusOr<std::map<uint64, int>> GetNodeIdToUnitMap() const
       SHARED_LOCKS_REQUIRED(chassis_lock);
 
+  ::util::StatusOr<int> GetUnitFromNodeId(uint64 node_id) const
+      SHARED_LOCKS_REQUIRED(chassis_lock);
+
   // Factory function for creating the instance of the class.
   static std::unique_ptr<BFChassisManager> CreateInstance(
       PhalInterface* phal_interface, BFPalInterface* bf_pal_interface);
@@ -110,9 +113,6 @@ class BFChassisManager {
 
   ::util::Status RegisterEventWriters() EXCLUSIVE_LOCKS_REQUIRED(chassis_lock);
   ::util::Status UnregisterEventWriters() LOCKS_EXCLUDED(chassis_lock);
-
-  ::util::StatusOr<int> GetUnitFromNodeId(uint64 node_id) const
-      SHARED_LOCKS_REQUIRED(chassis_lock);
 
   // Cleans up the internal state. Resets all the internal port maps and
   // deletes the pointers.

--- a/stratum/hal/lib/barefoot/bf_switch.cc
+++ b/stratum/hal/lib/barefoot/bf_switch.cc
@@ -217,7 +217,8 @@ BFSwitch::~BFSwitch() {}
                                        std::vector<::util::Status>* details) {
   absl::ReaderMutexLock l(&chassis_lock);
   for (const auto& req : request.requests()) {
-    ::util::StatusOr<DataResponse> resp;
+    DataResponse resp;
+    ::util::Status status = ::util::OkStatus();
     switch (req.request_case()) {
       case DataRequest::Request::kOperStatus:
       case DataRequest::Request::kAdminStatus:
@@ -226,9 +227,15 @@ BFSwitch::~BFSwitch() {}
       case DataRequest::Request::kPortCounters:
       case DataRequest::Request::kAutonegStatus:
       case DataRequest::Request::kFrontPanelPortInfo:
-      case DataRequest::Request::kLoopbackStatus:
-        resp = bf_chassis_manager_->GetPortData(req);
+      case DataRequest::Request::kLoopbackStatus: {
+        auto port_data = bf_chassis_manager_->GetPortData(req);
+        if (!port_data.ok()) {
+          status.Update(port_data.status());
+        } else {
+          resp = port_data.ConsumeValueOrDie();
+        }
         break;
+      }
       case DataRequest::Request::kNodeInfo: {
         auto* node_info = resp.mutable_node_info();
         node_info->set_vendor_name("Barefoot");
@@ -236,15 +243,18 @@ BFSwitch::~BFSwitch() {}
         break;
       }
       default:
-        // TODO(antonin)
-        resp = MAKE_ERROR(ERR_INTERNAL) << "Not supported yet";
+        status =
+            MAKE_ERROR(ERR_UNIMPLEMENTED)
+            << "Request type "
+            << req.descriptor()->FindFieldByNumber(req.request_case())->name()
+            << " is not supported yet: " << req.ShortDebugString() << ".";
         break;
     }
-    if (resp.ok()) {
+    if (status.ok()) {
       // If everything is OK send it to the caller.
-      writer->Write(resp.ValueOrDie());
+      writer->Write(resp);
     }
-    if (details) details->push_back(resp.status());
+    if (details) details->push_back(status);
   }
   return ::util::OkStatus();
 }

